### PR TITLE
Dockerfile rework

### PR DESCRIPTION
**Pull request by mutantcornholio**
* Negative approach to .dockerignore
* Reintruduction of turbo
* Removed unnecessary apt-get calls
* Moved docs compilation into a separate layer, which gets cached
* Changed turbo paths to actual output directories

### CODE REVIEW

**Comment by ironoa** [comment]
https://github.com/w3f/1k-validators-be-PRIVATE/issues/81

File: .dockerignore [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
-combined.log
-docker-compose.yml
-node_modules/
-
-*/**/node_modules/
\ No newline at end of file
+*
+!docs
+!packages
+!scripts
```
Comment (ironoa) [inline .dockerignore:4]: is scripts needed ?

File: Dockerfile-dev [hunk 2]
```diff
@@ -5,12 +5,12 @@ ARG PACKAGE
...
 WORKDIR /code
-RUN echo "building ${PACKAGE}... "; \
-    yarn install; \
-    echo "yarn install done. Building...." ; \
-    yarn build; \
-    echo "building ${PACKAGE} done."; \
+RUN echo "building ${PACKAGE}... " && \
+    yarn install && \
+    echo "yarn install done. Building...." && \
+    yarn build && \
```
Comment (ironoa) [inline Dockerfile-dev:11]: should this use turbo as well ?

**Comment by ironoa** [comment]
can we remove the building of the not used images (worker, ...) from the pipeline ?

File: Dockerfile-dev (hunk 2 above)
Comment (mutantcornholio) [reply Dockerfile-dev:11]: Not sure... 
What's the purpose of this Dockerfile anyway?  
It builds the package normal way, then runs `CMD yarn run start:dev:${PACKAGE}`, which is `node esbuild.js --watch & nodemon --watch build --exec node --max-old-space-size=8024 build/index.mjs`, which is watch mode compilation, so `yarn build` gets completely ignored...  
I also don't understand why somebody wants to run watchmode compilation in Docker environment.

**Comment by mutantcornholio** [comment]
> can we remove the building of the not used images (worker, ...) from the pipeline ?

done

File: Dockerfile-dev (hunk 2 above)
Comment (mutantcornholio) [reply Dockerfile-dev:11]: Will be addressed in https://github.com/w3f/1k-validators-be-PRIVATE/issues/84

**Pull request merged**